code/game/objects/items/stacks/stack.dm

```diff
@@ -235,6 +235,11 @@
 		O.color = color
 	use(R.req_amount * multiplier)
 
+	if(isitemstack(O))
+		var/obj/item/stack/stack = O
+		if(stack.merge_with_stack_in_hands(user))
+			return // Stack is merged and original item is deleted
```

Comment:
It's a niche case and probably fine anyway since none of the lower code looks relevant, but what about when a partial merge occurs due to the in hand stack being near cap? Say you're holding 45 and create 10 of something with a max 50 stack. Probably matters more for the comment assuming deletion than functionality, honestly.

Reply:
Good point, I'm gonna test it

Reply:
Tested with stacks of razor wire assembly (those craft 2 at a time). Seems like nothing breaks, but I changed code and removed early return